DifferenceGenerator/src/main/kotlin/DifferenceGenerator.kt

```diff
@@ -0,0 +1,58 @@
+import org.bytedeco.javacv.FFmpegFrameGrabber
+import java.awt.image.BufferedImage
+import java.io.File
+
+class DifferenceGenerator(video1Path: String, video2Path: String, outputPath: String) :
+    AbstractDifferenceGenerator(video1Path, video2Path, outputPath) {
+    val outputFile = File(outputPath)
+    val video1File = File(video1Path)
+    val video2File = File(video2Path)
+
+    /**
+     * Initializes a new instance of the [DifferenceGenerator] class.
+     *
+     * @throws Exception if the videos are not in an [AcceptedCodecs.ACCEPTED_CODECS].
+     */
+    init {
+        if (isLosslessCodec(video1File) && isLosslessCodec(video2File)) {
+            val video1Grabber = FFmpegFrameGrabber(video1File)
+            val video2Grabber = FFmpegFrameGrabber(video2File)
+
+            generateDifference(video1Grabber, video2Grabber)
+        } else {
+            throw Exception("Videos must be in a lossless codec")
+        }
+    }
+
+    /**
+     * Determines whether the given video file is encoded using one of the
+     * [AcceptedCodecs.ACCEPTED_CODECS].
+     *
+     * @param videoFile the video file to check
+     * @return true if the video file is encoded using one of the [AcceptedCodecs.ACCEPTED_CODECS],
+     * false otherwise
+     */
+    private fun isLosslessCodec(videoFile: File): Boolean {
+        val grabber = FFmpegFrameGrabber(videoFile)
+        grabber.start()
+        val codecName = grabber.videoMetadata["encoder"]
+        grabber.stop()
+        if (codecName == null) {
+            throw Exception("Video must have a codec")
+        }
+
+        for (codec in AcceptedCodecs.ACCEPTED_CODECS) {
+            if (codecName.contains(codec)) {
+                return true
+            }
+        }
+        return false
```

Comment:
use sets with the keyword `in`

Reply:
I'm not sure here, because names of encoders are often not strictly defined and vary between encoding softwares, thus an exact match will not always occur. 
If I use `if(codecName in AcceptedCodecs.ACCEPTED_CODECS)` some encoders will be missed. 
For example the video I encoded has the encoder names `Uncompressed YUV 422 10-bit`.
On the other hand I see that my approach is also a bit shaky.

Reply:
The longer I think about it the dirtier my solution seems. So I guess it makes sense to hardcode the exact encoders that are accepted. Thanks for the good review btw :)